test/cypress/tests/modules/BlockEvents/Backspace.cy.ts

```diff
@@ -0,0 +1,380 @@
+    cy.createEditor({
+      data: {
+        blocks: [
+          {
+            id: 'block1',
+            type: 'paragraph',
+            data: {
+              text: '', // empty block
+            },
+          },
+          {
+            id: 'block2',
+            type: 'paragraph',
+            data: {
+              text: 'Not empty block',
+            },
+          },
+        ],
+      },
+    }).as('editorInstance');
+
+    cy.get('[data-cy=editorjs]')
+      .find('.ce-paragraph')
+      .last()
+      .click()
+      .type('{home}') // move caret to the beginning
+      .type('{backspace}');
+
+    cy.get<EditorJS>('@editorInstance')
+      .then(async (editor) => {
+        const { blocks } = await editor.save();
+
+        expect(blocks.length).to.eq(1); // one block has been removed
+        expect(blocks[0].id).to.eq('block2'); // second block is still here
+      });
+  });
+
+  it('should remove current Block if it is empty, but previous is not. Also, should close the Toolbox and set Caret to the end of the prev Block', function () {
+    cy.createEditor({
+      data: {
+        blocks: [
+          {
+            id: 'block1',
+            type: 'paragraph',
+            data: {
+              text: 'Not empty block',
+            },
+          },
+          {
+            id: 'block2',
+            type: 'paragraph',
+            data: {
+              text: '',  // empty block
+            },
+          },
+        ],
+      },
+    }).as('editorInstance');
+
+    cy.get('[data-cy=editorjs]')
+      .find('.ce-paragraph')
+      .last()
+      .click()
+      .type('{backspace}');
+
+    /**
+     * Current Block has been removed
+     */
+    cy.get<EditorJS>('@editorInstance')
+      .then(async (editor) => {
+        const { blocks } = await editor.save();
+
+        expect(blocks.length).to.eq(1); // one block has been removed
+        expect(blocks[0].id).to.eq('block1'); // second block is still here
+      });
+
+    /**
+     * Caret is set to the end of the previous Block
+     */
+    cy.window()
+      .then((window) => {
+        const selection = window.getSelection();
+        const range = selection.getRangeAt(0);
+
+        cy.get('[data-cy=editorjs]')
+          .find('.ce-paragraph')
+          .should(($block) => {
+            expect($block[0].contains(range.startContainer)).to.be.true;
+            expect(range.startOffset).to.be.eq($block[0].textContent.length);
+          });
+      });
+
+    /**
+     * Toolbox has been closed
+     */
+    cy.get('[data-cy=editorjs]')
+      .find('.ce-toolbar')
+      .should('not.have.class', 'ce-toolbar--opened');
+  });
+
+  it('should merge current Block with the previous one if Caret at the start of the Block and both Blocks are mergeable. Also, should close the Toolbox. Caret should be places in a place of glue', function () {
```

Comment:
Just thinking about cases when there are two mergeable blocks with different type. Which type you'll leave for backspace and for delete events? Maybe add test cases?

Reply:
There will be a target block type. 

`mergeBlock` just rerenders it with new data. It should be covered by Block Manager unit test, but currently we don't have unit tests. I think, it would be covered in Document Model.

Reply:
I was wrong, currently we can merge only similar types of blocks. I thought that we're using our `ConversionConfig` feature for merging, but seems like it does not implemented yet. 

I'll add a statement for `type` checking for now. Later I'll implement Conversion, and remove the `type` restriction. And we'll have an ability to merge any blocks.